docs/Rendszerterv.md

```diff
@@ -93,14 +93,152 @@ Vendég vásárló
+
+2. Webszerver réteg
+
+Nginx vagy Apache: Kiszolgálja az alkalmazást, kezeli a bejövő kéréseket, biztosítja a statikus fájlok (képek, stíluslapok) gyors elérését, és HTTPS protokollon keresztül titkosítja a kapcsolatot.
+SSL-tanúsítvány: Biztosítja az összes adat biztonságos átvitelét a kliens és a webszerver között.
+
+3. Alkalmazás réteg (Laravel)
+
+MVC architektúra:
+Modellek: Az adatbázissal kapcsolatos műveleteket kezelik. Például a felhasználók, LEGO termékek és rendelési tételek modelljei biztosítják az ORM alapú adatműveleteket.
+Vezérlők (Controllers): Kezeli a felhasználói műveleteket (pl. bejelentkezés, termékkezelés, rendelésmódosítás). Az egyes végpontokhoz (URL-ekhez) külön vezérlők kapcsolódnak, amelyek a megfelelő logikát biztosítják.
+Nézetek (Views): Az admin és vásárlói nézetek különálló felületként jelennek meg. A Blade sablonmotor biztosítja a dinamikus HTML-generálást.
+Köztes réteg (Middleware):
+Autentikáció és jogosultság kezelés (admin és vásárlói jogosultságok elhatárolása).
+CSRF védelem és token alapú biztonság biztosítása.
+
+4. Adatbázis réteg
+
+MySQL vagy PostgreSQL: Tárolja az összes tartalmat és kapcsolatot (felhasználók, termékek, rendelési adatok).
+Adatbázis struktúra:
+Táblák a felhasználók, termékek (LEGO-k), rendelések, rendelési tételek, és felhasználói címek tárolására.
+Külső kulcsok és indexek az adatok közötti gyors és hatékony kapcsolatokhoz.
+
+5. Fájltároló réteg
+
+Helyi tárhely vagy felhő alapú tárhely (pl. AWS S3): Profilképek és egyéb statikus fájlok biztonságos és hatékony tárolására.
+Laravel Storage: A fájlkezelést egyszerűsíti, valamint hozzáférést biztosít a különböző fájltárolási megoldásokhoz.
+
+6. Gyorsítótárazási réteg
+
+Redis vagy Memcached: Az adatbázis műveletek gyorsítása érdekében a gyakran használt adatokat (pl. készlet információk) cache-ben tárolja.
+Session kezelés: Az autentikációs adatok gyors elérése és a felhasználói munkamenetek kezelése cache használatával.
+
+7. API réteg
+
+RESTful API végpontok: Biztosítják az integrációt egy esetleges mobilalkalmazással vagy más külső szolgáltatással. Lehetővé teszi a termékinformációk, rendelési műveletek API-k keresztüli elérését.
+Autentikáció és jogosultság kezelés: Token alapú hitelesítési megoldás (pl. Laravel Passport vagy Sanctum) az API végpontok biztonságos eléréséhez.
+
+8. Adminisztrációs és üzemeltetési eszközök
+
+CI/CD Pipeline: Az alkalmazás verziókezeléséhez és folyamatos integrációjához (GitLab CI, GitHub Actions) automatizált telepítési folyamat.
+Monitoring és Loggolás: A rendszer figyelése és a hibakeresés érdekében eszközök (pl. Laravel Telescope, Sentry, Grafana) biztosítják az alkalmazás teljesítményének és hibáinak nyomon követését.
+Biztonsági mentés: Adatbázis és fájl biztonsági mentés rendszeres időközönként, a felhasználói adatok védelmére.
+
+
 ### A program elérése, kezelése
 
+interneten
+
 ## 9. Adatbázis terv
```

Comment:
Ezt az adatbázis tervet jobb lenne valami diagrammal ábrázolni és nem leírni.  Az is jó ha kiscreenshotoljátok a phpmyadmin felületről